src/main/java/pl/llp/aircasting/screens/stream/NoteViewerActivity.java

```diff
@@ -0,0 +1,157 @@
+package pl.llp.aircasting.screens.stream;
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.content.FileProvider;
+import android.view.View;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.google.inject.Inject;
+
+import java.io.File;
+
+import pl.llp.aircasting.Intents;
+import pl.llp.aircasting.R;
+import pl.llp.aircasting.model.Note;
+import pl.llp.aircasting.screens.common.base.DialogActivity;
+import pl.llp.aircasting.screens.common.base.SimpleProgressTask;
+import pl.llp.aircasting.screens.common.helpers.FormatHelper;
+import pl.llp.aircasting.screens.common.helpers.PhotoHelper;
+import pl.llp.aircasting.screens.common.sessionState.VisibleSession;
+import pl.llp.aircasting.storage.repository.NoteRepository;
+
+import static pl.llp.aircasting.screens.stream.base.AirCastingActivity.NOTE_INDEX;
+import static pl.llp.aircasting.Intents.triggerSync;
+
+public class NoteViewerActivity extends DialogActivity implements View.OnClickListener {
+    @Inject VisibleSession mVisibleSession;
+    @Inject PhotoHelper mPhotoHelper;
+    @Inject NoteRepository mNotesRepository;
+
+    private View mPreviousNote;
+    private View mNextNote;
+    private TextView mNoteDate;
+    private TextView mNoteNumber;
+    private TextView mNoteText;
+    private Button mNoteSave;
+    private Button mNoteDelete;
+    private Button mViewPhoto;
+
+    private Note mCurrentNote;
+    private int mNotesTotal = 0;
+    private int mNoteIndex = -1;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.note_viewer);
+
+        mPreviousNote = findViewById(R.id.note_left);
+        mNextNote = findViewById(R.id.note_right);
+        mNoteDate = findViewById(R.id.note_date);
+        mNoteText = findViewById(R.id.note_text);
+        mNoteNumber = findViewById(R.id.note_number);
+        mViewPhoto = findViewById(R.id.view_photo);
+        mNoteDelete = findViewById(R.id.note_delete);
+        mNoteSave = findViewById(R.id.note_save);
+
+        mNextNote.setOnClickListener(this);
+        mPreviousNote.setOnClickListener(this);
+        mViewPhoto.setOnClickListener(this);
+        mNoteSave.setOnClickListener(this);
+        mNoteDelete.setOnClickListener(this);
+
+        mNotesTotal = mVisibleSession.getSessionNoteCount();
+        mNoteIndex = getIntent().getIntExtra(NOTE_INDEX, 0);
+
+        setNote();
+    }
+
+    private void setNote() {
+        int index = ((mNoteIndex % mNotesTotal) + mNotesTotal) % mNotesTotal;
```

Comment:
Could you elaborate on the math here please?

Reply:
Kinda doesn't make sense, does it :D Nothing tricky here, just me copy-pasting and skimming over it

Reply:
Have you considered a comment or extracting a function to make it a bit more clear?

Reply:
I could, but since this basically boils down to `index = mNoteIndex % mNotesTotal` if I'm reading this correctly, I'll probably just leave it in this simpler form, since it's pretty self-explanatory that way.